rs/pocket_ic_server/src/pocket_ic.rs

```diff
@@ -831,6 +843,233 @@ impl Operation for GetTime {
+    }
+
+    fn id(&self) -> OpId {
+        OpId("get_canister_http".into())
+    }
+}
+
+// START COPY from rs/https_outcalls/client/src/client.rs
+
+#[derive(Clone)]
+pub struct SingleResponseAdapter {
+    response: Result<CanisterHttpSendResponse, (Code, String)>,
+}
+
+impl SingleResponseAdapter {
+    fn new(response: Result<CanisterHttpSendResponse, (Code, String)>) -> Self {
+        Self { response }
+    }
+}
+
+#[tonic::async_trait]
+impl CanisterHttpService for SingleResponseAdapter {
+    async fn canister_http_send(
+        &self,
+        _request: Request<CanisterHttpSendRequest>,
+    ) -> Result<Response<CanisterHttpSendResponse>, Status> {
+        match self.response.clone() {
+            Ok(resp) => Ok(Response::new(resp)),
+            Err((code, msg)) => Err(Status::new(code, msg)),
+        }
+    }
+}
+
+async fn setup_adapter_mock(
+    adapter_response: Result<CanisterHttpSendResponse, (Code, String)>,
+) -> Channel {
+    let (client, server) = tokio::io::duplex(1024);
+    let mock_adapter = SingleResponseAdapter::new(adapter_response);
+    tokio::spawn(async move {
+        Server::builder()
+            .add_service(CanisterHttpServiceServer::new(mock_adapter))
+            .serve_with_incoming(futures::stream::iter(vec![Ok::<_, std::io::Error>(server)]))
+            .await
+    });
+
+    let mut client = Some(client);
+    Endpoint::try_from("http://[::]:50051")
+        .unwrap()
+        .connect_with_connector(service_fn(move |_: Uri| {
+            let client = client.take();
+
+            async move {
+                if let Some(client) = client {
+                    Ok(client)
+                } else {
+                    Err(std::io::Error::new(
+                        std::io::ErrorKind::Other,
+                        "Client already taken",
+                    ))
+                }
+            }
+        }))
+        .await
+        .unwrap()
+}
+
+// END COPY
+
+fn process_mock_canister_https_response(
+    pic: &PocketIc,
+    mock_canister_http_response: &MockCanisterHttpResponse,
+) -> OpOut {
+    let subnet_id =
+        ic_types::SubnetId::new(ic_types::PrincipalId(mock_canister_http_response.subnet_id));
+    let subnet = pic.get_subnet_with_id(subnet_id);
+    match subnet {
+        Some(subnet) => {
+            let canister_http_request_id =
+                CanisterHttpRequestId::from(mock_canister_http_response.request_id);
+            let contexts = subnet.canister_http_request_contexts();
+            if let Some(context) = contexts.get(&canister_http_request_id) {
+                let timeout = context.time + Duration::from_secs(5 * 60);
+                let canister_id = context.request.sender;
+                let content = match &mock_canister_http_response.response {
+                    CanisterHttpResponse::CanisterHttpReply(reply) => {
+                        let grpc_channel = pic.runtime.block_on(setup_adapter_mock(Ok(
```

Comment:
grpc?

Reply:
Sure but does it really conform to the google RPC standard? Or does this abbreviation also mean something else?

Reply:
This is how it's called in the production component: https://github.com/dfinity/ic/blob/d13d7748207c5e7be26114fdec7abdb9913e527e/rs/https_outcalls/client/src/client.rs#L57

Reply:
ok, not important then.